Classes/Networking/RequestBody/TMGZIPEncodedRequestBody.m

```diff
@@ -0,0 +1,119 @@
+//
+//  TMGZIPEncodedRequestBody.m
+//  TMTumblrSDK
+//
+//  Created by Michael Benedict on 2/9/18.
+//
+
+#import "TMRequestBody.h"
+#import "TMGZIPEncodedRequestBody.h"
+#include <zlib.h>
+#define CHUNKSIZE (1024*4)
+
+@interface TMGZIPEncodedRequestBody ()
+@property (nonatomic, nonnull, readonly) id<TMRequestBody> originalBody;
+@property (nonatomic, nonnull, readonly) NSData *compressedBodyData;
+-(NSData*)compressedBody:(NSData *)data;
+@end
+
+@implementation TMGZIPEncodedRequestBody
+
+- (nonnull instancetype)initWithRequestBody:(nonnull id<TMRequestBody>)body {
+    NSParameterAssert(body);
+    self = [super init];
+    
+    if (self) {
+        _originalBody = body;
+        _compressedBodyData = [self compressedBody:[body bodyData]];
```

Comment:
Are these objects ever allocated on the main thread? Is this method sufficiently fast enough to do on that if so?

Reply:
its up to the caller but probably should not be... it probably wouldn't be horrible if a smallish request body was requested on the main thread but one could certainly make a request big enough that it would lock up the main thread for quite awhile

Reply:
Makes sense, yeah 👍